logtape/logger.ts

```diff
@@ -526,21 +527,50 @@ export class LoggerImpl implements Logger {
       },
       timestamp: Date.now(),
       properties: {},
-    });
+    };
+
+    let cachedProps: Record<string, unknown> | undefined = undefined;
+    Object.assign(
+      record,
+      typeof properties === "function"
+        ? {
+          get properties() {
+            if (cachedProps == null) cachedProps = properties();
+            return cachedProps;
+          },
+        }
+        : { properties },
+    );
```

Comment:
According to Codecov, this code is not covered by tests, could you please add a test to cover this part?